aidial_sdk/utils/streaming.py

```diff
@@ -60,23 +77,90 @@ async def to_block_response(stream: ResponseStream) -> dict:
 
 
 async def to_streaming_response(
-    stream: ResponseStream,
+    stream: ResponseStreamWithStr,
```

Comment:
Have you considered to incorporate raw streaming events into our chunks?

Something like this
```python

class EventStreamCompatible(ABC):
    @abstractmethod
    def to_event_stream_chunk(self) -> str:
        pass


class BaseChunk(ABC, EventStreamCompatible):
    @abstractmethod
    def to_dict(self) -> Dict[str, Any]:
        pass

    # Move _format_chunk here
    def to_event_stream_chunk(self) -> str:
        data = "data: " + (
            json.dumps(data, separators=(",", ":"))
            if isinstance(data, dict)
            else data
        )
        log_debug(data)
        return f"{data}\n\n"

```

Reply:
In such way you can implement both heartbeat and `_DONE_MARKER` as `EventStreamCompatible` implementations

Reply:
I'm not sure. Chunks and Stream elements exist on two different logical levels.

Chunks are JSON parts of the response.

A stream element is a more generic thing - whatever I could encode in SSE response, this includes various infra/markup things like [DONE] and the heartbeat.

Reply:
Yeah, that's why I introduced hierarchy, where `EventStreamCompatible` do no rely on `BaseChunk`, but vice versa

We do know for sure that main purpose of chunks - is to be used inside of stream, so I think this realtion is logical
`EventStreamCompatible` - everything, that implements method, that returns string in event-stream-compatible format
`BaseChunk` - some JSON response, that will be used inside of event stream

Reply:
But ok, maybe let's traverse back to this idea, when we'll have more chunks, that are non-dicts